OsmAnd-java/src/main/java/net/osmand/binary/BinaryMapPoiReaderAdapter.java

```diff
@@ -778,7 +786,16 @@ private Amenity readPoiPoint(int left31, int right31, int top31, int bottom31,
 			case OsmandOdb.OsmAndPoiBoxDataAtom.SUBCATEGORIES_FIELD_NUMBER:
 				int subtypev = codedIS.readUInt32();
 				retValue.setLength(0);
+				hasSubcategoriesField = true;
 				PoiSubType st = region.getSubtypeFromId(subtypev, retValue);
+				if (req.poiTopIndexAdditionalFilter != null) {
```

Comment:
poiTopIndexAdditionalFilter -> poiAdditionalFilter (make it work for non top index ? )